hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaConverter.java

```diff
@@ -49,12 +49,14 @@
    */
   public static HCatSchema toHiveSchema(Schema schema, HCatSchema tableSchema) {
     List<HCatFieldSchema> fields = Lists.newArrayList();
+    //TODO: Finalize a consistent way to whether do projection on source or not while reading and if needed
+    // do type check here
     for (Schema.Field field : schema.getFields()) {
       String name = field.getName();
       try {
         // this field of the schema must exist in the table
         Preconditions.checkNotNull(tableSchema.get(name), "Missing field %s in table schema", name);
-        fields.add(new HCatFieldSchema(name, getType(name, field.getSchema().getType()), ""));
+        fields.add(new HCatFieldSchema(name, getType(name, field.getSchema()), ""));
```

Comment:
Rather than creating the TypeInfo from the field schema, should just take whatever is already in the tableSchema since that is guaranteed to be correct.  Also need to perform validation to catch mismatches between the passed in Schema vs the actual HCatSchema